vault/plugin_catalog.go

```diff
@@ -168,6 +170,7 @@ func (c *Core) setupPluginCatalog(ctx context.Context) error {
 		directory:       c.pluginDirectory,
 		logger:          c.logger,
 		mlockPlugins:    c.enableMlock,
+		wrapper:         logical.StaticSystemView{VersionString: version.GetVersion().Version},
```

Comment:
This is questionable, it makes the tests pass but should we be assuming a static (vs dynamic) system view?

Reply:
I think static is generally used for tests. However, we do use it to shim the system view when a plugin is in metadata mode. Maybe this is ok though for the purposes of the plugin catalog?

Reply:
I think it's ok, but I don't know a whole lot about this part of the code.

Reply:
I guess it's ok because any plugin that relies on PluginEnv to get the version will be invoking that on a dynamic system view, at some point.  Whereas this is used solely for the check to see if GRPC is supported.